packages/router/src/lib/meta-tags.ts

```diff
@@ -0,0 +1,99 @@
+import { inject } from '@angular/core';
+import { Meta, MetaDefinition as NgMetaTag } from '@angular/platform-browser';
+import { ActivatedRouteSnapshot, NavigationEnd, Router } from '@angular/router';
+import { filter } from 'rxjs/operators';
+
+export const ROUTE_META_TAGS_KEY = Symbol(
+  '@analogjs/router Route Meta Tags Key'
+);
+
+const CHARSET_KEY = 'charset';
+const HTTP_EQUIV_KEY = 'httpEquiv';
+// httpEquiv selector key needs to be in kebab case format
+const HTTP_EQUIV_SELECTOR_KEY = 'http-equiv';
+const NAME_KEY = 'name';
+const PROPERTY_KEY = 'property';
+const CONTENT_KEY = 'content';
+
+export type MetaTag =
+  | (CharsetMetaTag & ExcludeRestMetaTagKeys<typeof CHARSET_KEY>)
+  | (HttpEquivMetaTag & ExcludeRestMetaTagKeys<typeof HTTP_EQUIV_KEY>)
+  | (NameMetaTag & ExcludeRestMetaTagKeys<typeof NAME_KEY>)
+  | (PropertyMetaTag & ExcludeRestMetaTagKeys<typeof PROPERTY_KEY>);
+
+type CharsetMetaTag = { [CHARSET_KEY]: string };
+type HttpEquivMetaTag = { [HTTP_EQUIV_KEY]: string; [CONTENT_KEY]: string };
+type NameMetaTag = { [NAME_KEY]: string; [CONTENT_KEY]: string };
+type PropertyMetaTag = { [PROPERTY_KEY]: string; [CONTENT_KEY]: string };
+
+type MetaTagKey =
+  | typeof CHARSET_KEY
+  | typeof HTTP_EQUIV_KEY
+  | typeof NAME_KEY
+  | typeof PROPERTY_KEY;
+type ExcludeRestMetaTagKeys<Key extends MetaTagKey> = {
+  [K in Exclude<MetaTagKey, Key>]?: never;
+};
+
+type MetaTagSelector =
+  | typeof CHARSET_KEY
+  | `${
+      | typeof HTTP_EQUIV_SELECTOR_KEY
+      | typeof NAME_KEY
+      | typeof PROPERTY_KEY}="${string}"`;
+type MetaTagMap = Record<MetaTagSelector, MetaTag>;
+
+export function updateMetaTagsOnNavigationEnd(): void {
+  const router = inject(Router);
+  const metaService = inject(Meta);
+
+  router.events
+    .pipe(filter((event) => event instanceof NavigationEnd))
+    .subscribe(() => {
+      const metaTagMap = getMetaTagMap(router.routerState.snapshot.root);
```

Comment:
If they visit a route with meta tags then one without, are the previous tags left over?

Reply:
Yes!

I'd keep it as it is to avoid additional DOM manipulations. I think the most important is to have valid meta tags when a specific page is initially loaded by the browser or any other app.

What do you think? 👀 

Btw, there is the same behavior with Angular `title` - if we don't provide a new title for a route, the previous title will remain when that route is activated.

Reply:
I agree with that behavior along with the router title also.